### PR TITLE
BUG: set_precision format fixed (#13257)

### LINKED ISSUE #13257: Pandas style.set_precision not working as expected
I expected the following code to give me a representation of my data that is rounded or truncated to 2 decimals.

```
import pandas as pd
import numpy as np

df = pd.DataFrame(data = np.random.randn(4,4), columns=['A','B','C','D'])
df.style.set_precision(2)
```

However, it will always return 2 digits instead, so for example, I want 2.37419 to be displayed as 2.37, but instead this code will give me 2.4. Am I overseeing something or is this a bug?

Note that the following code works as I expected, but then I change a general setting whereas I would like to apply it only to the current output:

```
pd.set_option('display.precision',2)
df
```

I'm using pandas version 0.18.1 in Jupyter notebook on Firefox, with python 2.7.11.final.0 and IPython 4.1.2 on 64-bits linux.

### PR DESCRIPTION
**Pull request by AntonioAndraues**
- [x] closes #13257
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

I've tried to fix the set_precision format, does that fix it? 

### INPUT -> df.style.set_precision(2)
| A | B |
| --- | --- |
| 1 | 2.0090 |
| 3.2121 | 4.566 |
### OUTPUT
| A | B |
| --- | --- |
| 1 | 2.01 |
| 3.21 | 4.57 |

@vinigl
@samuelgranato

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Can you add a test for this

**Comment by AntonioAndraues** [comment]
> Can you add a test for this

I just added a test.
Is that it?

**Review by TomAugspurger — COMMENTED** [review]
Looks good. Can you add a release note to doc/source/whatsnew/v1.0.0.rst under "bug fixes"? Probably the "Other" section.

**Comment by AntonioAndraues** [comment]
> Looks good. Can you add a release note to doc/source/whatsnew/v1.0.0.rst under "bug fixes"? Probably the "Other" section.

Done

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -201,7 +201,7 @@ ExtensionArray
 Other
 ^^^^^
 - Trying to set the ``display.precision``, ``display.max_rows`` or ``display.max_columns`` using :meth:`set_option` to anything but a ``None`` or a positive int will raise a ``ValueError`` (:issue:`23348`)
-
+- Bug in pandas io formats style : fixed function default_display_func because wasn`t displaying decimals digits as supposed (:issue:`13257`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.0.0.rst]: refer to the method on styler. :meth:\`...\` 

"wasn't" should use an apostrophe, not a backtick.

Comment (AntonioAndraues) [reply doc/source/whatsnew/v1.0.0.rst]: My bad...
fixed

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -201,7 +201,7 @@ ExtensionArray
 Other
 ^^^^^
 - Trying to set the ``display.precision``, ``display.max_rows`` or ``display.max_columns`` using :meth:`set_option` to anything but a ``None`` or a positive int will raise a ``ValueError`` (:issue:`23348`)
-
+- Bug in :meth:`pandas.io.formats.style.Styler.default_display_func` because wasn't displaying decimals digits as supposed (:issue:`13257`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.0.0.rst]: Sorry, I misread this earlier. `default_display_func` isn't a method on Styler, it's a closure.

Rather, just describe the behavior and change. Something like

```suggestion
- Bug in :meth:`pandas.io.formats.style.Styler` formatting for floating values not displaying decimals correctly (:issue:`13257`)
```

File: pandas/io/formats/style.py [hunk 3]
```diff
@@ -151,7 +151,8 @@ def __init__(
 
         def default_display_func(x):
             if is_float(x):
-                return "{:>.{precision}g}".format(x, precision=self.precision)
+                display_format = "{0:.{precision}f}".format(x, precision=self.precision)
+                return display_format.rstrip("0").rstrip(".")
```
Comment (simonjayhawkins) [inline pandas/io/formats/style.py]: what is the reason for this?

why is the output not consistent with the standard terminal output?

```python
>>>
>>> import pandas as pd
>>>
>>> pd.__version__
'0.25.0+227.g347ad8564'
>>>
>>> df = pd.DataFrame(data=[[1.0, 2.0090], [3.2121, 4.566]], columns=["a", "b"])
>>>
>>> with pd.option_context("precision", 1):
...     print(df)
...
     a    b
0  1.0  2.0
1  3.2  4.6
>>> with pd.option_context("precision", 2):
...     print(df)
...
      a     b
0  1.00  2.01
1  3.21  4.57
>>> with pd.option_context("precision", 3):
...     print(df)
...
       a      b
0  1.000  2.009
1  3.212  4.566
>>>

```

Comment (simonjayhawkins) [reply pandas/io/formats/style.py]: my understanding of https://github.com/pandas-dev/pandas/pull/13897#issuecomment-237551157 is that ints do not have decimal points and trailing zeros but that floats should follow the precision setting @TomAugspurger ?

The comments also states that "Ideally df.style would have the same output as df._repr_html() by default."

image

Comment (AntonioAndraues) [reply pandas/io/formats/style.py]: @simonjayhawkins  
The modification to make the program works as the way u said above is fair simple ( only remove rstrip ).
I really wanna just have the verdict of @TomAugspurger to make some last changes to the pull request.

Comment (TomAugspurger) [reply pandas/io/formats/style.py]: I'm not sure what's best here, but matching the output of the main DataFrame repr is probably best.

Comment (AntonioAndraues) [reply pandas/io/formats/style.py]: > I'm not sure what's best here, but matching the output of the main DataFrame repr is probably best.

Ok will match it now!

Comment (AntonioAndraues) [reply pandas/io/formats/style.py]: Just changed it. 
Now its working the same way as DataFrame repr is.
@TomAugspurger 

Captura de Tela 2019-09-03 às 08 51 30

**Comment by TomAugspurger** [comment]
Thanks. Looks like there's a merge conflict in 1.0.0.rst, if you could fix that.

**Comment by AntonioAndraues** [comment]
> Thanks. Looks like there's a merge conflict in 1.0.0.rst, if you could fix that.

I think i fix it.

**Comment by TomAugspurger** [comment]
CI is failing now. Do your tests need to be updated?

**Comment by AntonioAndraues** [comment]
> CI is failing now. Do your tests need to be updated?
@TomAugspurger 
I'm sorry forgot to update them... 
I think everything will be fine now!

**Comment by TomAugspurger** [comment]
Hmm CI is still failing. Haven't taken a look though.

**Comment by AntonioAndraues** [comment]
> Hmm CI is still failing. Haven't taken a look though.
I think CI is failing because some test from others contributors are expecting the older use of style...
Do i change the test?
Captura de Tela 2019-09-19 às 09 34 27

**Comment by WillAyd** [comment]
If the old test is wrong then for sure should update

**Comment by WillAyd** [comment]
@AntonioAndraues is this still active? If so can you fix up merge conflict?

**Comment by AntonioAndraues** [comment]
@WillAyd Yes. I Will fix it...
Sunday in the afternoon i commit it.

File: pandas/tests/io/formats/test_style.py [hunk 4]
```diff
@@ -1157,28 +1157,50 @@ def test_display_format_raises(self):
         with pytest.raises(TypeError):
             df.style.format(True)
 
+    def test_display_set_precision(self):
+        df = pd.DataFrame(data=[[1.0, 2.0090], [3.2121, 4.566]], columns=["a", "b"])
```
Comment (jreback) [inline pandas/tests/io/formats/test_style.py]: can you add the issue number here as a comment

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. small changes; ping on green.

**Review by jreback — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Great thanks @AntonioAndraues - nice PR!

**Comment by simonjayhawkins** [comment]
I _think_ this PR should be identified as a breaking change.

```python
import pandas as pd
pd.__version__
df = pd.DataFrame([0.1])
df
df.style.highlight_min()
```
in pandas 0.25.1
image

on master
image

**Pull request merged**